modules/apps/commerce/commerce-product-type-virtual-test/src/testIntegration/java/com/liferay/commerce/product/type/virtual/test/CPDefinitionVirtualTest.java

```diff
@@ -0,0 +1,172 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.type.virtual.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.commerce.product.model.CommerceCatalog;
+import com.liferay.commerce.product.service.CPDefinitionLocalService;
+import com.liferay.commerce.product.service.CommerceCatalogLocalServiceUtil;
+import com.liferay.commerce.product.test.util.CPTestUtil;
+import com.liferay.commerce.product.type.virtual.constants.VirtualCPTypeConstants;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.transaction.Propagation;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PersistenceTestRule;
+import com.liferay.portal.test.rule.TransactionalTestRule;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+
+import org.frutilla.FrutillaRule;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Christian Chiappa
+ */
+@RunWith(Arquillian.class)
+public class CPDefinitionVirtualTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(), PersistenceTestRule.INSTANCE,
+			new TransactionalTestRule(
+				Propagation.REQUIRED,
+				"com.liferay.commerce.product.type.virtual.service"));
+
+	@Before
+	public void setUp() throws Exception {
+		_commerceCatalog = CommerceCatalogLocalServiceUtil.addCommerceCatalog(
+			null, RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			LocaleUtil.US.getDisplayLanguage(),
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Iterator<CPDefinition> iterator = _cpDefinitions.iterator();
+
+		while (iterator.hasNext()) {
+			CPDefinition cpDefinitionToDelete = iterator.next();
+
+			_cpDefinitionLocalService.deleteCPDefinition(
+				cpDefinitionToDelete.getCPDefinitionId());
+
+			iterator.remove();
+		}
+	}
+
+	@Test
+	public void testCreate() throws Exception {
+		frutillaRule.scenario(
+			"Add product definition"
+		).given(
+			"I add a product definition"
+		).when(
+			"ignoreSKUCombinations is true"
+		).and(
+			"hasDefaultInstance is true"
+		).and(
+			"shippable is true"
```

Comment:
Why you don't add a clause for virtual product in this scenario?

Reply:
The test is inside a class that is called VirtualTest, so I thought is was not necessary to specify it. I can add it if you want

Reply:
Got it, but it's better to add it, thanks!